src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,202 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private Node<T> currentNode;
+    private Node<T> previousNode;
+    private Node<T> nextNode;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> newElement = new Node<T>(value);
+            head = newElement;
+            tail = newElement;
+        } else {
+            Node<T> newElement = new Node<T>(value);
```

Comment:
When creating a new Node<T>, pass all the fields to the Node constructor. Consider passing 'null' for 'prev' and 'next' if they are not needed.